virtcontainers/kata_agent.go

```diff
@@ -1550,6 +1556,61 @@ func (k *kataAgent) handleEphemeralStorage(mounts []specs.Mount) []*grpc.Storage
 	return epheStorages
 }
 
+// handleHugePages handles hugepages storage by
+// creating a Storage from corresponding source of the mount point
+func (k *kataAgent) handleHugepages(mounts []specs.Mount, hugepageLimits []specs.LinuxHugepageLimit) []*grpc.Storage {
+	//Map to hold the total memory of each type of hugepages
+	optionsMap := make(map[string]string)
+	var HugePageSizeUnitList = []string{"B", "K", "M", "G", "T", "P"}
+
+	for _, hp := range hugepageLimits {
+		if hp.Limit != 0 {
+			k.Logger().WithField("Pagesize", hp.Pagesize).Info("hugepage request")
+			k.Logger().WithField("Limit", hp.Limit).Info("hugepage request")
+			//example Pagesize 2MB, 1GB etc. The Limit are in Bytes
+			pageSize, _ := units.RAMInBytes(hp.Pagesize)
+			pageSizeStr := units.CustomSize("%g%s", float64(pageSize), 1024.0, HugePageSizeUnitList)
+			totalHpSize := fmt.Sprintf("%v", hp.Limit)
+			optionsMap[pageSizeStr] = totalHpSize
+		}
+	}
+
+	var hugepages []*grpc.Storage
+	for idx, mnt := range mounts {
+		//HugePages mount Type is Local
+		if mnt.Type == KataLocalDevType {
```

Comment:
This code is starting to get quite long and deeply indented. I'd consider inverting the logic to simplify it:

```go
if mnt.Type != KataLocalDevType {
    continue
}

if _, fsType, fsOptions, _ := ...
```

Reply:
Sure. Will change it